projects/alembic/alembic_dump_info_fuzzer.cc

```diff
@@ -0,0 +1,263 @@
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
+
+void dumpCurvesAttributes(const ICurvesSchema &schema) {
+  const size_t propertyCount = schema.getNumProperties();
+
+  for (size_t p = 0; p < propertyCount; p++) {
+    const PropertyHeader &header = schema.getPropertyHeader(p);
+    const PropertyType pType = header.getPropertyType();
+    const std::string &name = header.getName();
+
+    if (name == "P") {
+      schema.getNumSamples();
+    } else if (name == "N") {
+      schema.getNormalsParam().getNumSamples();
+    } else if (name == "uv" || name == "st") {
+      schema.getUVsParam().getNumSamples();
+    } else if (name == ".arbGeomParams") {
+      // additional geometry elements (color sets, additional texture
+      // coordinates)
+      const ICompoundProperty geoParam = schema.getArbGeomParams();
+      const size_t geoPropCount = geoParam.getNumProperties();
+
+      for (size_t g = 0; g < geoPropCount; g++) {
+        const PropertyHeader &headerGeo = geoParam.getPropertyHeader(g);
+        const std::string &nameGeo = headerGeo.getName();
+      }
+    }
+  }
+}
+
+void dumpPolyMesh(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IPolyMesh mesh(node.getParent(), header.getName());
+  const IPolyMeshSchema &schema = mesh.getSchema();
+
+  // Mesh properties
+  dumpMeshAttributes(schema);
+}
+
+void dumpSubD(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ISubD mesh(node.getParent(), header.getName());
+  const ISubDSchema &schema = mesh.getSchema();
+
+  dumpSubDAttributes(schema);
+  schema.getSubdivisionSchemeProperty();
+  schema.getFaceVaryingInterpolateBoundaryProperty();
+  schema.getFaceVaryingPropagateCornersProperty();
+  schema.getInterpolateBoundaryProperty();
+}
+
+void dumpFaceSet(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IFaceSet faceSet(node.getParent(), header.getName());
+  const IFaceSetSchema &schema = faceSet.getSchema();
+  schema.getNumSamples();
+}
+
+void dumpCurves(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ICurves curves(node.getParent(), header.getName());
+  const ICurvesSchema &schema = curves.getSchema();
+
+  dumpCurvesAttributes(schema);
+}
+
+void dumpXform(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IXform xform(node.getParent(), header.getName());
+  const IXformSchema &schema = xform.getSchema();
+
+  schema.getNumSamples();
+  schema.getNumOps();
+}
+
+void dumpMaterial(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  IMaterial material(node.getParent(), header.getName());
+  IMaterialSchema &schema = material.getSchema();
+
+  std::vector<std::string> targetNames;
+  size_t targetCount = targetNames.size();
+
+  for (size_t t = 0; t < targetCount; t++) {
```

Comment:
It's idiomatic C++ to use `std::iterator` (or `std::const_iterator`) to iterate over all of the collection types in the Standard Template Library.  Before the C++11 standard, this would lead to admittedly long-winded code like

```c++
for (std::const_iterator<std::string> it = targetNames.begin(); it != targetNames.end(); it++) {
  ...
}
```

Fortunately, C++11 introduced the syntactic sugar of the "range-based for loop", which lets you write

```c++
for (const std::string& target : targetNames) {
  ...
}
```

which you should use here.  You can even use the `auto` keyword (`for (const auto& tgt : targetNames)`) to make it a bit shorter.